pkg/externalauthprovider/flags.go

```diff
@@ -324,6 +324,21 @@ func CreateExternalAuthConfig(args *ExternalAuthProvidersArgs) (*cmv1.ExternalAu
 	externalAuthBuilder := cmv1.NewExternalAuth().ID(args.name)
 	claimValidationRules := args.claimValidationRule
 
+	// check parameters
+	if args.name == "" {
+		return &cmv1.ExternalAuth{}, fmt.Errorf(
+			"name parameter is mandatory for creating an external authentication configuration")
+	}
+
+	if args.issuerUrl == "" {
+		return &cmv1.ExternalAuth{}, fmt.Errorf(
+			"issuer-url parameter is mandatory for creating an external authentication configuration")
+	}
+	if args.issuerAudiences == nil {
+		return &cmv1.ExternalAuth{}, fmt.Errorf(
+			"issuer-audience parameter is mandatory for creating an external authentication configuration")
+	}
+
 	if args.issuerUrl != "" || args.issuerCaFile != "" || args.issuerAudiences != nil {
```

Comment:
Deleted my last comment to explain better;

Would just like to bring up something that may could be changed,

`if args.issuerUrl != "" || args.issuerCaFile != "" || args.issuerAudiences != nil {` --> we are checking the opposite conditions from 2 of the if statements above. For example:
`if args.issuerUrl == "" { return ... }`
There really isn't a need to have it in the condition below, because, if we get past this condition, then it is **always** not equal to an empty string.

Same goes for `if args.issuerAudience == nil { return ... }` --> The above if statement returns if it _is_ equal to nil, so there's no reason to check if it's not nil, as it will **always** be not nil after this if statement.

Not sure if it would be fine, given what you are trying to achieve, to only have `if args.issuerCaFile != "" { ... }` on L342, but just bringing this up :smile: LMK what you think of this

Reply:
overall, LGTM, just feels like redundant checks in that if statement due to the new if statements introduced

Reply:
Sounds good, @hunterkepley one question I got is that should I keep the current check parameters logic, or should I move them to condition1 || condition 2 || condition3. The reason I separated them is due to the error messages being different.

Reply:
I think it may be best to move inside the `condition1 || condition 2 || condition3` -- this way we run through the least amount of if statements in a best case scenario (no issues), while keeping about the same for worst case
WDYT?

Reply:
Sounds good to me, will also change that then. Thanks a lot @hunterkepley